ansible_collections/azureredhatopenshift/cluster/tasks/create_aro_cluster.yaml

```diff
@@ -0,0 +1,711 @@
+      register: az_extension_list
+      changed_when: false
+    - name: create_aro_cluster | Show az extensions
+      ansible.builtin.debug:
+        var: az_extension_list.stdout_lines
+    # - name: create_aro_cluster | Download azaroext
+    #   ansible.builtin.get_url:
+    #     url: https://arosvc.blob.core.windows.net/azext/aro-{{ AZAROEXT_VERSION }}-py2.py3-none-any.whl
+    #     dest: /tmp/aro-{{ AZAROEXT_VERSION }}-py2.py3-none-any.whl
+    #     mode: "0644"
+    #   delegate: "{{ delegation }}"
+    #   register: download_aroazext
+    # - name: create_aro_cluster | Install azaroext
+    #   when: download_aroazext is changed
+    #   ansible.builtin.command:
+    #     argv:
+    #       - az
+    #       - extension
+    #       - add
+    #       - -s="/tmp/aro-{{ AZAROEXT_VERSION }}-py2.py3-none-any.whl"
+
+# TODO: Create clusters with azure_rm_openshiftmanagedcluster if possible
+- name: create_aro_cluster | Custom service principal
+  when: create_csp | d(False)
+  block:
+    - name: create_aro_cluster | Create cluster service principal
+      ansible.builtin.command:
+        argv: [
+          "az", "ad", "sp", "create-for-rbac",
+          "-n", "{{ name }}-{{ resource_group }}-sp",
+          "--role", "contributor",
+          "--scopes", "{{ rg_info.state.id }}",
+          "-o=yaml"
+        ]
+      delegate_to: localhost
+      register: create_sp_output
+    - name: create_aro_cluster | Set fact csp_info
+      when: create_sp_output is success
+      ansible.builtin.set_fact:
+        csp_info: "{{ create_sp_output.stdout | from_yaml }}"
+    - name: create_aro_cluster | Show service principal
+      ansible.builtin.debug:
+        var: csp_info
+      tags:
+        - verbose
+
+# - name: Create cluster service principal
+#   azure.azcollection.azure_rm_adserviceprincipal:
+#     app_id: "{{ name | to_uuid }}"
+#     tenant: "{{ sub_info.tenant_id }}"
+#   delegate_to: localhost
+#   register: csp_info
+# - ansible.builtin.debug: var=csp_info
+# - name: Create cluster service principal
+#   ansible.builtin.command:
+#     delegate_to: localhost
+#     argv: [
+#       "az", "ad", "sp", "create-for-rbac",
+#       "-n", "{{ name }}-sp",
+#       "--role", "contributor",
+#       "--scopes", "{{ rg_info.state.id }}"
+#     ]
+#   register: sp_info
+# - ansible.builtin.debug: var=sp_info
+# - name: Create aro cluster
+#   azure.azcollection.azure_rm_openshiftmanagedcluster:
+#     delegate_to: localhost
+#     name: "{{ name }}"
+#     resource_group: "{{ resource_group }}"
+#     location: "{{ location }}"
+#     service_principal_profile:
+#       client_id: "{{ csp_info.state.client_id }}"
+#       client_secret: "{{ csp_info.state.client_secret }}"
+#     worker_profiles:
+#       - name: "worker"
+#         vm_size: "Standard_D4s_v3"
+#         subnet_id: "{{ worker_subnet_state.state.id }}"
+#     master_profile:
+#       vm_size: "Standard_D4s_v3"
+#       subnet_id: "{{ master_subnet_state.state.id }}"
```

Comment:
Should this be removed?
If it's WIP for later implementation, I think make a separate branch and removing it from this branch is ideal.